### PR TITLE
Some fixes for @angular/core

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: packages/core/src/view/provider.ts [hunk 1]
```diff
@@ -355,6 +355,12 @@ export function resolveDep(
   }
   const tokenKey = depDef.tokenKey;
 
+  if (tokenKey === ChangeDetectorRefTokenKey) {
+    // directives on the same element as a component should be able to control the change detector
+    // of that component as well.
+    allowPrivateServices = elDef && !!elDef.element !.componentView;
```
Comment (vicb) [inline packages/core/src/view/provider.ts]: normalize to boolean

File: packages/core/test/linker/view_injector_integration_spec.ts [hunk 2]
```diff
@@ -652,6 +652,46 @@ export function main() {
...
+            '<div componentNeedsChangeDetectorRef><div directiveNeedsChangeDetectorRef></div></div>');
+        cf.detectChanges();
+        const compEl = cf.debugElement.children[0];
+        const comp = compEl.injector.get(PushComponentNeedsChangeDetectorRef);
+        const dirEl = compEl.children[0];
+        const dir = dirEl.injector.get(DirectiveNeedsChangeDetectorRef);
+        comp.counter = 1;
+        cf.detectChanges();
+        expect(compEl.nativeElement).toHaveText('0');
+        dir.changeDetectorRef.markForCheck();
```
Comment (vicb) [inline packages/core/test/linker/view_injector_integration_spec.ts]: what is `dir.changeDetectorRef` in this case ?

File: packages/core/test/linker/regression_integration_spec.ts [hunk 3]
```diff
@@ -365,6 +366,23 @@ function declareTests({useJit}: {useJit: boolean}) {
...
+      @NgModule({declarations: [App], entryComponents: [App]})
+      class MyModule {
+      }
+
+      const modRef = TestBed.configureTestingModule({imports: [MyModule]})
+                         .get(NgModuleRef) as NgModuleRef<MyModule>;
+      const compRef =
+          modRef.componentFactoryResolver.resolveComponentFactory(App).create(Injector.NULL);
+
+      expect(getDOM().getAttribute(compRef.location.nativeElement, 'ng-version')).toBeFalsy();
```
Comment (vicb) [inline packages/core/test/linker/regression_integration_spec.ts]: you want `hasAttribute` here, no ?

File: packages/core/test/linker/view_injector_integration_spec.ts (hunk 2 above)
Comment (tbosch) [inline packages/core/test/linker/view_injector_integration_spec.ts]: It's the `changeDetectorRef` that the directive received. The test verifies that it cannot change mark the component on the parent element as being dirty.

File: packages/core/src/view/provider.ts (hunk 1 above)
Comment (tbosch) [inline packages/core/src/view/provider.ts]: Done

File: packages/core/test/linker/regression_integration_spec.ts (hunk 3 above)
Comment (tbosch) [inline packages/core/test/linker/regression_integration_spec.ts]: Done

**Review by vicb — APPROVED** [review]

**Pull request closed without merging**